### ISSUE TITLE
nvim-treesitter\parser\html.so is not a valid Win32

### ISSUE THREAD

**Issue by kunchenguid** [nvim-treesitter/nvim-treesitter#3484]
### Describe the bug

I'm running 64bit Neovim on Windows, using 64bit mingw/llvm/zig, yet still I'm getting the error "nvim-treesitter\parser\*.so is not a valid Win32 application".

### To Reproduce

1. Install Neovim from https://github.com/neovim/neovim/releases/nightly
2. Install LLVM through "choco install llvm"
3. Set require 'nvim-treesitter.install'.compilers = { 'clang' }
4. :TSInstall html
5. :checkhealth nvim-treesitter
6. See the error

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.7 (b268e412ad4848380166af153300464e5a1cf83f) (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.17.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `clang` executable found. Selected from { "clang" }
    Version: clang version 15.0.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - json           ✓ ✓ ✓ ✓ .
  - javascript     ✓ ✓ ✓ ✓ ✓
  - jsdoc          ✓ . . . .
  - typescript     ✓ ✓ ✓ ✓ ✓
  - markdown       x . x . x
  - php            x x x x x
  - python         x x x x x
  - yaml           x x x x x
  - css            ✓ . ✓ ✓ ✓
  - html           x x x x x
  - bash           x x x . x
  - lua            ✓ ✓ ✓ ✓ ✓
  - tsx            ✓ ✓ ✓ ✓ ✓
  - graphql        ✓ . . ✓ ✓
  - vim            ✓ ✓ ✓ . ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

## The following errors have been detected:
  - ERROR: markdown(highlights): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\markdown.so is not a valid Win32 application.
    
    markdown(highlights) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\markdown\highlights.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\markdown.so is not a valid Win32 application.
  - ERROR: markdown(folds): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\markdown.so is not a valid Win32 application.
    
    markdown(folds) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\markdown\folds.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\markdown.so is not a valid Win32 application.
  - ERROR: markdown(injections): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\markdown.so is not a valid Win32 application.
    
    markdown(injections) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\markdown\injections.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\markdown.so is not a valid Win32 application.
  - ERROR: php(highlights): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
    
    php(highlights) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\php\highlights.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
  - ERROR: php(locals): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
    
    php(locals) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\php\locals.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
  - ERROR: php(folds): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
    
    php(folds) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\php\folds.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
  - ERROR: php(indents): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
    
    php(indents) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\php\indents.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
  - ERROR: php(injections): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
    
    php(injections) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\php\injections.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\php.so is not a valid Win32 application.
  - ERROR: python(highlights): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
    
    python(highlights) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\python\highlights.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
  - ERROR: python(locals): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
    
    python(locals) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\python\locals.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
  - ERROR: python(folds): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
    
    python(folds) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\python\folds.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
  - ERROR: python(indents): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
    
    python(indents) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\python\indents.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
  - ERROR: python(injections): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
    
    python(injections) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\python\injections.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\python.so is not a valid Win32 application.
  - ERROR: yaml(highlights): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
    
    yaml(highlights) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\yaml\highlights.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
  - ERROR: yaml(locals): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
    
    yaml(locals) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\yaml\locals.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
  - ERROR: yaml(folds): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
    
    yaml(folds) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\yaml\folds.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
  - ERROR: yaml(indents): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
    
    yaml(indents) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\yaml\indents.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
  - ERROR: yaml(injections): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
    
    yaml(injections) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\yaml\injections.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\yaml.so is not a valid Win32 application.
  - ERROR: html(highlights): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
    
    html(highlights) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\html\highlights.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
    
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\html_tags\highlights.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
  - ERROR: html(locals): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
    
    html(locals) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\html\locals.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
  - ERROR: html(folds): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
    
    html(folds) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\html\folds.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
  - ERROR: html(indents): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
    
    html(indents) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\html\indents.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
    
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\html_tags\indents.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
  - ERROR: html(injections): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
    
    html(injections) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\html\injections.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\html.so is not a valid Win32 application.
  - ERROR: bash(highlights): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
    
    bash(highlights) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\bash\highlights.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
  - ERROR: bash(locals): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
    
    bash(locals) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\bash\locals.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
  - ERROR: bash(folds): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
    
    bash(folds) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\bash\folds.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
  - ERROR: bash(injections): Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
    
    bash(injections) is concatenated from the following files:
    | [ERROR]:"C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\queries\bash\injections.scm", failed to load: Failed to load parser: uv_dlopen: C:\Users\Kun\AppData\Local\nvim-data\site\pack\packer\start\nvim-treesitter\parser\bash.so is not a valid Win32 application.
```

### Output of `nvim --version`

```text
NVIM v0.8.0-dev-1070-g9b4cab012
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compiled by runneradmin@fv-az15-4

Features: -acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files (x86)/nvim/share/nvim"

Run :checkhealth for more info
```

### Additional context

I've tried mingw, zig and llvm. Uninstalled all parsers and reinstalled them each time. All getting the same errors. Any help or pointers for how to investigate further is appreciated!

**Comment by theHamsta** [comment]
> C:/Program Files (x86)/nvim/share/nvim"

Can you double check that this is really a 64bit installation (though it has its runtime dir in x86)?

**Comment by kunchenguid** [comment]
@theHamsta Is there a reliable way to check it at runtime? 

Here's where my nvim.exe comes from:
```
[11:59:41] ~ on❯ gcm nvim

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     nvim.exe                                           0.0.0.0    C:\Program Files\Neovim\bin\nvim.exe
```

"C:\Program Files (x86)\nvim" actually doesn't even exist on my machine.

**Comment by kunchenguid** [comment]
Oh gosh I figured it out. Apparently, on a 64bit Windows, none of the compilers mentioned in https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support actually produces the right built by default. I had to download https://github.com/mstorsjo/llvm-mingw and then explicitly use "x86_64-w64-mingw32-clang" as my compiler instead of just "clang". 

Not sure if my Windows was somehow special or something. But if this is the case for 64bit windows by default, the Windows-support documentation page above could really use some clarification.
